### ISSUE TITLE
model id 2 not update with the gdrive file

### ISSUE THREAD

**Issue by brainfo** [Nellaker-group/happy#14]
Hi,
I was testing on the sample wsi.tiff with the example in your third section in the readme. It turned out that the current repo is set up with the base db cell_model id 2: cell_model_accuracy_0.8204.pt; However, in the gdrive, there's only cell_model_accuracy_0.8472.pt
Either should this accuracy 0.8472 model be added to the database or we should replace the id 2 model with it. Which one is used in the paper?

Best,

**Comment by brainfo** [comment]
And in the graph_inference.py the hdf5 it trys to locate is $'projects/placenta/results/embeddings/lab_{run_id}/wsi_{run_id}/run_{run_id}.hdf5'
However, my results with copying of the cell_inference code from you read me is "projects/placenta/results/embeddings/lab_3/slide_sample_wsi.tif/run_4.hdf5"

I understand that it might be I didn't give the sample wsi an id to be wsi_{run_id}, but the lab_ is with 3 and run is with 4; is this something that can be fixed in the code?

**Comment by brainfo** [comment]
And for people who are not familiar with qupath like me, a script to load the annotation file in the qupath is valuable.

**Comment by brainfo** [comment]
And for the nuclei and cell type prediction (annotation), do we also get some resulted files? 

The analysis/visualisation/vis_nuclei_preds.py is only for the training dataset?

I don't know where are the output files from the nuclei and cell predict step (while the hdf5 file is obvious)

**Comment by cvanea** [comment]
Hi brainfo, 

Sorry for the delay in getting back to you! 

Thanks for spotting the model name discrepancy, the model used in the paper is the one on google drive, so 0.8472. The earlier model is from before peer review and I must’ve forgotten to update the demo db, my mistake!

I am a little confused about the problem here, this seems to be correct: "projects/placenta/results/embeddings/lab_3/slide_sample_wsi.tif/run_4.hdf5”. Could you elaborate further? Is it not able to find the hdf5 file?

Regarding loading into QuPath, you can do it through the UI. Once you have the wsi open, click on the tool at the top with three dots. That will open the ‘counting window’ which will have a button called ‘Load points’. Once you click that navigate to the tsv file that you want to open and this will load all points into QuPath. If you’re not already familiar with QuPath it’s definitely worth learning! It’s an awesome tool. There’s a bit of a learning curve but it’s great for generating training data and exploring model outputs. 

Nuclei and cell inference outputs go into both the db and an hdf5 file for that run but there are no other output files. A tsv file of nuclei coordinates and cell predictions can be extracted from the db using qupath/coord_to_tsv.py if you want to visualise them in QuPath. 

Cell and tissue id mappings can be found in the dataclass in /happy/organs.py. That’s also where you would extend the models to add new classes or another organ. 

The dots discrepancy between nuclei and tissues is due to a post-processing step on syncytial knots, if --group-knts is kept True which is the default. The knots with more than 4 nuclei are merged into one point and any isolated knot predictions are renamed as syncytiotrophoblasts. To match the cell and tissue predictions you need to join on x and y coordinates and exclude those not present in the tissue tsv. The plotting scripts under /projects/placenta/plots also take --group-knts and a --trained-with-grouped-knts argument to handle this. I appreciate that this is somewhat fiddly!

The tissue tsv output file from graph_inference.py could be changed to save the post-processed cell predictions from line 67. This is cleaner but now the tissue tsv can’t be loaded into QuPath and the plotting scripts would need updating to reflect this new column.

I hope that helps!
